### PR TITLE
Httpd request context

### PR DESCRIPTION
**Pull request by mmaslankaprv**
Added per listener TLS setup (fixed recognizing protocol schema) and ability to recognize which listener accepted given connection. This feature is needed for the Kafka-Rest compatibility proxy in redpanda as we need to return a contextual information based on HTTP or HTTPS as per the protocol description and need underlying access to the actual listener information.

### CODE REVIEW

**Review by BenPope — APPROVED** [review]

File: include/seastar/http/httpd.hh [hunk 1]
```diff
@@ -201,7 +205,7 @@ public:
     // RFC 7231, Section 7.1.1.1.
     static sstring http_date();
 private:
-    future<> do_accept_one(int which);
+    future<> do_accept_one(int which, bool encrypted);
```
Comment (senior7515) [inline include/seastar/http/httpd.hh:208]: `bool tls=false`

File: src/http/httpd.cc [hunk 2]
```diff
@@ -498,20 +513,20 @@ future<> http_server::stop() {
 }
 
 // FIXME: This could return void
-future<> http_server::do_accepts(int which) {
-    (void)try_with_gate(_task_gate, [this, which] {
-        return keep_doing([this, which] {
-            return try_with_gate(_task_gate, [this, which] {
-                return do_accept_one(which);
+future<> http_server::do_accepts(int which, bool encrypted) {
```
Comment (senior7515) [inline src/http/httpd.cc:516]: change `encrypted=tls`

File: include/seastar/http/request.hh [hunk 3]
```diff
@@ -137,6 +137,10 @@ struct request {
         return content_type_class == ctclass::app_x_www_urlencoded;
     }
 
+    int get_listener_idx() const {
```
Comment (senior7515) [inline include/seastar/http/request.hh:140]: add docstrings

File: include/seastar/http/httpd.hh [hunk 4]
```diff
@@ -75,11 +75,12 @@ class connection : public boost::intrusive::list_base_hook<> {
     queue<std::unique_ptr<reply>> _replies { 10 };
     bool _done = false;
     bool _encrypted;
+    int _listener_idx;
```
Comment (senior7515) [inline include/seastar/http/httpd.hh:78]: looks great. I'd add on the cover letter that this is needed for the Kafka-Rest compatibility proxy in redpanda as we need to return a contextual information based on HTTP or HTTPS as per the protocol description and need underlying access to the actual listener information.

**Review by senior7515 — COMMENTED** [review]
minor nits. looks good.

**Review by dotnwat — APPROVED** [review]
lgtm. it might be worth changing encrypted to tls as Alex mentioned. I could see someone on Seastar-dev bringing up that naming.

**Pull request closed without merging**